src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -487,9 +490,62 @@ EventEngine::TaskHandle PosixEventEngine::RunAfterInternal(
   return handle;
 }
 
+#if GRPC_ARES == 1 && defined(GRPC_POSIX_SOCKET_TCP)
```

Comment:
Note for the future - something you already know, but worth commenting - that this engine is expected to run on platforms that don't have Ares enabled (Android is the only supported platform I'm aware of anymore if iOS works with Ares, but possibly other platforms will use it well). Before the EE dns functionality can be enabled generally, a basic A/AAAA-only native resolver needs to be implemented as well.

It's probably worth having a TODO for that here or below next to the `Crash("unimplemented")` bit. If I missed a TODO for this somewhere already, apologies.